src/realm/error_codes.cpp

```diff
@@ -50,19 +49,27 @@ ErrorCategory ErrorCodes::error_categories(Error code)
         case SubscriptionFailed:
         case UnsupportedFileFormatVersion:
         case OperationAborted:
+            return ErrorCategory().set(ErrorCategory::runtime_error);
+
         case AutoClientResetFailed:
+        case BadChangeset:
         case ConnectionClosed:
+        case SyncProtocolInvariantFailed:
```

Comment:
I think these are being kept in alphabetical order, would you please move `SyncProtocolInvariantFailed` back to where it was.